### PR TITLE
feat: Technical review: Add docs for relative CSS colors

### PR DESCRIPTION
**Pull request by chrisdavidmills**
Note: This technical review is now completed and approved. For the follow-on editorial review, see https://github.com/mdn/content/pull/32004

---

### Description

[Chrome 119](https://chromestatus.com/feature/5205844613922816) and [Safari 16.4](https://webkit.org/blog/13966/webkit-features-in-safari-16-4/#color) add [relative color syntax](https://drafts.csswg.org/css-color-5/#relative-colors) for CSS `<color>` types.

This PR adds docs to cover this new feature.

See my [research document](https://docs.google.com/document/d/1YjI97ZrRF2qtI5KUV_TiG3OVmyJQ2TBqosqKGxy1Dhw/edit) for more details of what this work entails.

### CODE REVIEW

**Review by estelle — COMMENTED** [review]
i took a look at the hsl page. i realize it's a draft form, but figured early feedback would be helpful.

File: files/en-us/web/css/color_value/hsl/index.md [hunk 1]
```diff
@@ -18,29 +18,76 @@ Defining _complementary colors_ with `hsl()` can be done with a single formula,
 ## Syntax
 
 ```css
+/* Absolute values */
 hsl(120deg 75% 25%)
 hsl(120deg 75% 25% / 0.6)
+
+/* Relative values */
+hsl(from red 240 s l) /* transform red to blue */
```
[the remark this answers is not in the document]
Comment (chrisdavidmills) [reply files/en-us/web/css/color_value/hsl/index.md:26]: I've gone a bit further with this, adding a bit more description to each of the relative examples and moving the longer comments onto their own line in each case. Better descriptions are good here, as the relative syntax is a bit hard to grok when you first meet it.

File: files/en-us/web/css/color_value/hsl/index.md [hunk 2]
```diff
@@ -18,29 +18,76 @@ Defining _complementary colors_ with `hsl()` can be done with a single formula,
...
 > **Note:** See [Missing color components](/en-US/docs/Web/CSS/color_value#missing_color_components) for the effect of `none`.
 
+#### Relative values
+
+Functional notation: `hsl(from color h s l[ / alpha])`
+
+- `from`
+  - : The keyword `from` is always included when defining a relative color.
+- `color`
+  - : A {{cssxref("&lt;color&gt;")}} value representing the **originating color** that the relative color is based on. Note that this can be _any_ valid color value, including another relative color.
```
[the remark this answers is not in the document]
Comment (chrisdavidmills) [reply files/en-us/web/css/color_value/hsl/index.md:57]: I wrestled with this a bit, and eventually settled on

> - : A {{cssxref("&lt;color&gt;")}} value representing the **originating color** that the relative color is based on. Note that this can be _any_ valid {{cssxref("&lt;color&gt;")}} syntax, including another relative color.

File: files/en-us/web/css/color_value/hsl/index.md [hunk 3]
```diff
@@ -18,29 +18,76 @@ Defining _complementary colors_ with `hsl()` can be done with a single formula,
...
+- `h`
+  - : A {{CSSXref("&lt;number&gt;")}}, an {{CSSXref("&lt;angle&gt;")}}, or the keyword `none`, which represents the hue angle of the output color.
```
[the remark this answers is not in the document]
Comment (chrisdavidmills) [reply files/en-us/web/css/color_value/hsl/index.md:59]: Couple of things here:

- `none` was already explained with a link to a section giving more details in the absolute syntax section. I've added some more details about it in the page before giving the link to more info. I've also moved this below both sections, as it is relevant to absolute and relative syntax.
- It is not as simple as saying "the letter `h` can represent hue". Ultimately the output hue value is set as a number or angle, but the variable `h` (provided by the browser, which equals the hue of the originating color) can be used when defining the output color channels. You could set the output hue to be `h` (same as the originating color), or `240` (absolute value) or `calc(h + 120)` (value relative to the originating color hue). If you wanted to, you could even set the one, two, or all three output channel values to `h`. This wouldn't make sense, but you can still do it.

I therefore decided to explain the letters separately, below.

File: files/en-us/web/css/color_value/hsl/index.md [hunk 4]
```diff
@@ -18,29 +18,76 @@ Defining _complementary colors_ with `hsl()` can be done with a single formula,
...
+```css
+hsl(from red h s l)
+```
+
+This function uses none of the originating color's channel values, and outputs a completely different color not based on it:
+
+```css
+hsl(from red 240 60 70)
+```
+
```
[the remark this answers is not in the document]
Comment (chrisdavidmills) [reply files/en-us/web/css/color_value/hsl/index.md:80]: The first two examples you give here don't work in Chrome or Safari. As I've just discovered, they've implemented calculations inside relative colors in incompatible ways! 

To provide support across both engines, you'd need to use both of these:

```css
/* Saf - requires units on all values */
hsl(from red calc(h + 240deg) calc(s - 5%) calc(l - 10%)) 

/* Chrome - requires no units in calculations, and weird decimal percentage equivalents for percentages */
hsl(from red calc(h + 240) calc(s - 0.5) calc(l - 0.1)) 
```

I've written this mess up in my next commit, and also adjusted the examples a bit.

File: files/en-us/web/css/color_value/hsl/index.md [hunk 5]
```diff
@@ -18,29 +18,76 @@ Defining _complementary colors_ with `hsl()` can be done with a single formula,
...
+The following function use some of the originating color's channel values inside {{cssxref("calc()")}} functions to calculate new channel values for the output color:
+
+```css
+hsl(from red calc(h + 60) calc(s - 0.2) calc(l - 0.1))
+```
+
+> **Note:** When expressing absolute percentage values, e.g. for the `s` and `l` channels, you can express them as whole numbers or percentages (e.g. `60` or `60%` will represent "sixty percent"). However, when expressing relative percentage values based on the originating color's channel values, you need to express the modifiers as decimal values (e.g. `calc(s - 0.2)` means "subtract 20% for the originating color's saturation value").
```
[the remark this answers is not in the document]
Comment (chrisdavidmills) [reply files/en-us/web/css/color_value/hsl/index.md:87]: This bit has changed quite a bit now. In my next commit, there is a whole separate section about weird browser differences.

File: files/en-us/web/css/color_value/hsl/index.md [hunk 6]
```diff
@@ -146,6 +193,60 @@ div.hsla {
 
 {{EmbedLiveSample("legacy_syntax_hsla", "100%", 150)}}
 
+### Using relative colors with hsl()
+
+The below example styles three {{htmlelement("div")}} elements with different background colors. The middle one is given the unmodified `--base-color`, while the left and right ones are given lightened and darkened variants of that `--base-color`.
+
+These variants are defined using relative colors — the `--base-color` is passed into an `hsl()` function, and the output color has its `l` (lightness) channel modified to achieve the desired effect via a `calc()` function. The lightened color has 20% added to the lightness channel, and the darkened color has 20% subtracted from it.
```
[the remark this answers is not in the document]
Comment (chrisdavidmills) [reply files/en-us/web/css/color_value/hsl/index.md:200]: I have mostly made these changes. More explanation has been added earlier about `0`/`none`, although as I said before, the explanation of the `h`, `s`, and `l` values is not that simple. My next commit should hopefully clear this up a bit.

File: files/en-us/web/css/color_value/index.md [hunk 7]
```diff
@@ -51,6 +51,11 @@ oklab(59% 0.1 0.1 / 0.5)
 oklch(60% 0.15 50)
 oklch(60% 0.15 50 / 0.5)
 
+/* Relative CSS colors */
+rgb(from red 0 0 255) /* transform red to blue */
```
[the remark this answers is not in the document]
Comment (chrisdavidmills) [reply files/en-us/web/css/color_value/index.md:55]: No, this outputs pure blue. The three channel values are set to absolute values of 0, 0, and 255.

File: files/en-us/web/css/css_colors/relative_colors/index.md [hunk 8]
```diff
@@ -0,0 +1,745 @@
...
+## Color functions that support relative colors
+
+In the section above we only saw relative colors defined via the [`rgb()`](/en-US/docs/Web/CSS/color_value/rgb) function. However, relative colors can be defined using any modern CSS function — [`color()`](/en-US/docs/Web/CSS/color_value/color), [`hsl()`](/en-US/docs/Web/CSS/color_value/hsl), [`hwb()`](/en-US/docs/Web/CSS/color_value/hwb), [`lab()`](/en-US/docs/Web/CSS/color_value/lab), [`lch()`](/en-US/docs/Web/CSS/color_value/lch), [`oklab()`](/en-US/docs/Web/CSS/color_value/oklab), [`oklch()`](/en-US/docs/Web/CSS/color_value/oklch), or [`rgb()`](/en-US/docs/Web/CSS/color_value/rgb). The syntax structure is exactly the same in each case — the only thing that differs is the names of the provided originating color channel values.
+
+Below you can find relative color examples for each color function. Each case is the simplest possible, with the output channels exactly matching the originating color channels:
+
+```css
+/* color() with and without alpha channel */
+color(from red a98-rgb r g b)
+color(from red a98-rgb r g b / alpha)
```
Comment (romainmenke) [inline files/en-us/web/css/css_colors/relative_colors/index.md:118]: Maybe an example with `xyz`?

I think most authors will associate the available channels with the name of the function but that isn't true for `color()`.

File: files/en-us/web/css/color_value/hsl/index.md [hunk 9]
```diff
@@ -18,28 +18,127 @@ Defining _complementary colors_ with `hsl()` can be done with a single formula,
...
+hsl(from rgb(255 0 0) calc(h + 60deg) calc(s - 20%) calc(l - 10%)) /* Safari */
+```
+
+> **Note:** When expressing absolute percentage values, e.g. for the `s` and `l` channels, you can express them as whole numbers or percentages (e.g. `60` or `60%` will represent "sixty percent"). However,
+
+> **Note:** See [Using relative colors](/en-US/docs/Web/CSS/CSS_colors/Relative_colors) for a detailed guide to relative colors and their use cases, with more detailed examples than the ones available on this page.
+
+### Browser support differences
+
+There are currently some syntax differences between browsers for `hsl()`, which warrant some explanation. Chrome supports unitless values for percentages, and other browsers don't. It is therefore advised to specify the units:
```
Comment (romainmenke) [inline files/en-us/web/css/color_value/hsl/index.md:114]: This is not limited to `hsl()` and it is also untrue :)
`hsl(60deg 40 70)` does not work in current Chrome, will be fixed in the next release : `121`.

See : https://github.com/mdn/browser-compat-data/issues/21404

Maybe better to teach users about the relaxed syntax instead of temporary support differences?

File: files/en-us/web/css/css_colors/relative_colors/index.md [hunk 10]
```diff
@@ -0,0 +1,745 @@
...
+{{ EmbedLiveSample("Using math functions", "100%", "200") }}
+
+## Browser support differences
+
+There are currently some syntax differences between browsers for relative CSS colors, which warrant some explanation. Chrome supports unitless values for percentages, and other browsers don't. It is therefore advised to specify the units:
+
+```css
+hsl(from rgb(255 0 0) 60deg 40% 70%) /* Works in Chromiums and Safari */
+hsl(from rgb(255 0 0) 60deg 40 70) /* Works only in Chromiums */
+```
```
Comment (romainmenke) [inline files/en-us/web/css/css_colors/relative_colors/index.md:268]: This example is unrelated to relative color syntax.
So I think it is confusing to mention it here.

File: files/en-us/web/css/css_colors/relative_colors/index.md [hunk 11]
```diff
@@ -0,0 +1,745 @@
...
+/* Chromiums */
+background-color: hsl(
+  from rgb(255 0 0) calc(h + 60) calc(s - 0.2) calc(l - 0.1)
+);
+
+/* Safari */
+background-color: hsl(
+  from rgb(255 0 0) calc(h + 60deg) calc(s - 20%) calc(l - 10%)
+);
+```
```
Comment (romainmenke) [inline files/en-us/web/css/css_colors/relative_colors/index.md:287]: I am not sure if it is good to call out this difference in this way.

Safari shipped an initial implementation when the specification was still undergoing (breaking) changes.

Reading this I don't know which syntax is actually correct.

I think it is better to ignore this temporary compat pain and to just consider Safari's implementation as bugged/partial until it is fixed.

And to only document/teach to correct syntax to authors.

File: files/en-us/web/css/css_colors/relative_colors/index.md (hunk 8 above)
Comment (chrisdavidmills) [reply files/en-us/web/css/css_colors/relative_colors/index.md:118]: True, that could be a little confusing. In my next commit I've used `x y z`, as per your suggestion.

File: files/en-us/web/css/color_value/hsl/index.md (hunk 9 above)
Comment (chrisdavidmills) [reply files/en-us/web/css/color_value/hsl/index.md:114]: I've just deleted this section, and replaced it with a section that talks about the fact that channel values all resolve to `<number>` values. I think it makes more sense to talk about the relaxed syntax (eventually) in the color function pages rather than here.

File: files/en-us/web/css/css_colors/relative_colors/index.md (hunk 10 above)
Comment (chrisdavidmills) [reply files/en-us/web/css/css_colors/relative_colors/index.md:268]: Again, this got deleted

File: files/en-us/web/css/css_colors/relative_colors/index.md (hunk 11 above)
Comment (chrisdavidmills) [reply files/en-us/web/css/css_colors/relative_colors/index.md:287]: All deleted now. The docs will just teach the standards, and we'll talk about the compat problems just in the compat data (which we'll have to update again ;-) )

File: files/en-us/web/css/css_colors/relative_colors/index.md [hunk 12]
```diff
@@ -108,14 +111,14 @@ rgb(from rgb(200 170 0) b g r)
...
 
-Below you can find relative color examples for each color function. Each case is the simplest possible, with the output channels exactly matching the originating color channels:
+Below you can find relative color syntax examples for each color function. Each case is the simplest possible, with the output channels exactly matching the origin color channels:
 
 ```css
 /* color() with and without alpha channel */
-color(from red a98-rgb r g b)
-color(from red a98-rgb r g b / alpha)
+color(from red a98-rgb x y z)
+color(from red a98-rgb x y z / alpha)
```
Comment (romainmenke) [inline files/en-us/web/css/css_colors/relative_colors/index.md:121]:
```suggestion
color(from red a98-rgb r g b)
color(from red a98-rgb r g b / alpha)

color(from red xyz-d50 x y z)
color(from red xyz-d50 x y z / alpha)
```

It is either `r g b` or `x y z` depending on the color space.
But I understand the confusion :)
Apparently Chrome has an implementation bug and incorrectly accepts `color(from red a98-rgb x y z)` as a valid color.

see : https://github.com/web-platform-tests/wpt/pull/43877

File: files/en-us/web/css/css_colors/relative_colors/index.md (hunk 11 above)
Comment (romainmenke) [reply files/en-us/web/css/css_colors/relative_colors/index.md]: Thanks!

File: files/en-us/web/css/color_value/hsl/index.md (hunk 9 above)
Comment (romainmenke) [reply files/en-us/web/css/color_value/hsl/index.md]: I agree.

File: files/en-us/web/css/css_colors/relative_colors/index.md (hunk 8 above)
Comment (romainmenke) [reply files/en-us/web/css/css_colors/relative_colors/index.md]: I should have been clearer here :)

File: files/en-us/web/css/css_colors/relative_colors/index.md (hunk 12 above)
Comment (chrisdavidmills) [reply files/en-us/web/css/css_colors/relative_colors/index.md:121]: Ah, ha, I get it now. I'm not familiar with how `color()` works, so this is useful, thanks.

I'll accept your suggestion in a second. There still seem to be several bugs and things that could change with this feature. I might pause my docs work until it becomes more stable.

Comment (romainmenke) [reply files/en-us/web/css/css_colors/relative_colors/index.md]: > There still seem to be several bugs and things that could change with this feature. I might pause my docs work until it becomes more stable.

I think the remaining bugs are relatively minor and that good docs will actually help push this forward. If CSS authors learn through trial and error they might learn the wrong syntax by accident. But I defer to your judgement on this :)

Comment (chrisdavidmills) [reply files/en-us/web/css/css_colors/relative_colors/index.md]: Yeah, you could be right here. I'm going to do some detailed testing today to see if I uncover much other weirdness with relative colors in the different models.

File: files/en-us/web/css/css_colors/relative_colors/index.md [hunk 13]
```diff
@@ -0,0 +1,722 @@
...
+2. The syntax inside the color function is a bit different for relative colors. First of all, you pass in an **origin color** (represented above by _`origin-color`_) to base your relative color on, preceded by the `from` keyword. This can be any valid {{cssxref("&lt;color&gt;")}} value using any available color model, including a color value contained in a [CSS custom property](/en-US/docs/Web/CSS/Using_CSS_custom_properties).
+3. The individual channels of the output color are then defined after the origin color — represented above by the _`channel1`_, _`channel2`_, and _`channel3`_ placeholders. The channels defined here depend on the color function you are using for your relative color. If you are using `hsl()` for example, you would need to define hue, saturation, and lightness values. These could be static values, or values based on the channel values of the origin color.
+4. Optionally, an _`alpha`_ channel value for the output color is defined, preceded by a slash (`/`).
```
Comment (romainmenke) [inline files/en-us/web/css/css_colors/relative_colors/index.md:25]: https://drafts.csswg.org/css-color-5/#:~:text=If%20the%20alpha%20value%20is%20omitted%2C%20however%2C%20it%20defaults%20to%20taking%20from%20the%20origin%20color%20(rather%20than%20defaulting%20to%20100%25%2C%20as%20it%20does%20in%20the%20absolute%20syntax).

> If the alpha value is omitted, however, it defaults to taking from the [origin color](https://drafts.csswg.org/css-color-5/#origin-color) (rather than defaulting to 100%, as it does in the absolute syntax).

You can see this in action here : https://codepen.io/romainmenke/pen/QWoEzjK

Comment (chrisdavidmills) [reply files/en-us/web/css/css_colors/relative_colors/index.md:25]: I just noticed this comment. Details added in my next commit!

File: files/en-us/web/css/color_value/hsl/index.md [hunk 14]
```diff
@@ -18,28 +18,97 @@ Defining _complementary colors_ with `hsl()` can be done with a single formula,
...
+```css
+hsl(from rgb(255 0 0) h s l)
+```
+
+> **Note:** If the output color is using a different color model to the origin color, the origin color is converted to the same model as the output color in the background so that it can be represented in a way that is compatible (i.e. using the same channels).
+
+This function uses absolute values for the output color's channel values, outputting a completely different color not based on the origin color:
+
+```css
+hsl(from rgb(255 0 0) 240 60 70)
```
Comment (argyleink) [inline files/en-us/web/css/color_value/hsl/index.md:104]:
```suggestion
hsl(from rgb(255 0 0) 240 60% 70%)
```

File: files/en-us/web/css/color_value/lch/index.md [hunk 15]
```diff
@@ -30,10 +53,53 @@ Functional notation: `lch(L C H[ / A])`
...
+```css
+lch(from hsl(0 100% 50%) l c h)
+```
+
+> **Note:** If the output color is using a different color model to the origin color, the origin color is converted to the same model as the output color in the background so that it can be represented in a way that is compatible (i.e. using the same channels).
+
+This function uses absolute values for the output color's channel values, outputting a completely different color not based on the origin color:
+
+```css
+lch(from hsl(0 100% 50%) 29.6871 66.83 327.109)
```
Comment (argyleink) [inline files/en-us/web/css/color_value/lch/index.md:95]:
```suggestion
lch(from hsl(0 100% 50%) 29.6871% 66.83 327.109)
```

File: files/en-us/web/css/color_value/oklch/index.md [hunk 16]
```diff
@@ -30,7 +49,53 @@ Functional notation: `oklch(L C H[ / A])`
...
+```css
+oklch(from hsl(0 100% 50%) l c h)
+```
+
+> **Note:** If the output color is using a different color model to the origin color, the origin color is converted to the same model as the output color in the background so that it can be represented in a way that is compatible (i.e. using the same channels).
+
+This function uses absolute values for the output color's channel values, outputting a completely different color not based on the origin color:
+
+```css
+oklch(from hsl(0 100% 50%) 42.1 48.25 328.363)
```
Comment (argyleink) [inline files/en-us/web/css/color_value/oklch/index.md:91]: chroma here looks like a typo, that's way too high.

```suggestion
oklch(from hsl(0 100% 50%) 42.1% 0.25 328.363)
```

**Review by argyleink — COMMENTED** [review]
i hope these marked items are helpful 🙂

File: files/en-us/web/css/color_value/oklch/index.md (hunk 16 above)
Comment (chrisdavidmills) [reply files/en-us/web/css/color_value/oklch/index.md:91]: Yup, copy/paste error ;-)

File: files/en-us/web/css/color_value/hsl/index.md (hunk 14 above)
Comment (mysteryDate) [reply files/en-us/web/css/color_value/hsl/index.md]: Modern hsl syntax can use percentages _or_ numbers 
https://drafts.csswg.org/css-color-4/#the-hsl-notation

File: files/en-us/web/css/color_value/color/index.md [hunk 17]
```diff
@@ -14,27 +14,94 @@ Support for a particular colorspace can be detected with the [`color-gamut`](/en
...
 ```css
+/* Absolute values */
+
 color(display-p3 1 0.5 0);
 color(display-p3 1 0.5 0 / .5);
+
+/* Relative values */
+
+/* Change the r g and b channels to transform red to purple */
+color(from rgb(255 0 0) srgb-linear 0.215811 -1.07102e-8 0.215811)
```
Comment (mysteryDate) [inline files/en-us/web/css/color_value/color/index.md:25]: I don't understand this one, looks like you're not using hte relative origin color at all? i.e. this color is equivalent to
`color(srgb-linear 0.215811 0 0.215811))`

File: files/en-us/web/css/color_value/color/index.md [hunk 18]
```diff
@@ -14,27 +14,94 @@ Support for a particular colorspace can be detected with the [`color-gamut`](/en
...
 
 - `colorspace`
 
-  - : An {{CSSXref("&lt;ident&gt;")}} denoting one of the predefined color spaces: `srgb`, `srgb-linear`, `display-p3`, `a98-rgb`, `prophoto-rgb`, `rec2020`, `xyz`, `xyz-d50`, and `xyz-d65`.
+  - : An {{CSSXref("&lt;ident&gt;")}} denoting one of the predefined colorspaces: `srgb`, `srgb-linear`, `display-p3`, `a98-rgb`, `prophoto-rgb`, `rec2020`, `xyz`, `xyz-d50`, or `xyz-d65`.
 
 - `c1`, `c2`, `c3`
 
-  - : {{CSSXref("number")}} between 0 and 1, a {{CSSXref("percentage")}} or the keyword `none`, which provide the component values in the color space.
+  - : Each value can be written as a {{CSSXref("number")}} between 0 and 1, a {{CSSXref("percentage")}}, or the keyword `none`. These values represent the component values for the colorspace.
```
Comment (mysteryDate) [inline files/en-us/web/css/color_value/color/index.md:50]: So, the numbers are not bounded to [0,1]. [0,1] represent the bounds of the colorspace.
`color(srgb 0 1 0)` is srgb green and
`color(display-p3 0 1 0)` is display-3 green but
`color(srgb -0.51 1.02 -0.31)` is a totally valid color (it is in fact display-p3 green _in_ srgb)

File: files/en-us/web/css/color_value/color/index.md [hunk 19]
```diff
@@ -14,27 +14,94 @@ Support for a particular colorspace can be detected with the [`color-gamut`](/en
...
+For example, let's start with an origin color of `hsl(0 100% 50%)` (equivalent to `red`). The following functions output the same color as the origin color — they use the origin color's color channel values as the output channel values:
+
+```css
+color(from hsl(0 100% 50%) srgb r g b)
+color(from hsl(0 100% 50%) xyz x y z)
+```
+
+> **Note:** If the output color is using a different color model to the origin color, the origin color is converted to the same model as the output color in the background so that it can be represented in a way that is compatible (i.e. using the same channels).
+
+These functions use absolute values for the output color channel values, outputting completely different colors not based on the origin color:
```
Comment (mysteryDate) [inline files/en-us/web/css/color_value/color/index.md:92]: Why would you ever want to do this?

File: files/en-us/web/css/color_value/color/index.md [hunk 20]
```diff
@@ -187,6 +254,81 @@ div {
...
+  background-color: var(--base-color);
+}
+
+#three {
+  background-color: color(
+    from var(--base-color) display-p3 r calc(g - 0.15) calc(b - 0.15)
+  );
+}
+
+/* Use @supports to add in support for Safari 16.4+, which supports old
```
Comment (mysteryDate) [inline files/en-us/web/css/color_value/color/index.md:308]: Is there a bug for this on webkit?

File: files/en-us/web/css/color_value/color/index.md (hunk 17 above)
Comment (chrisdavidmills) [reply files/en-us/web/css/color_value/color/index.md:25]: I suppose I wanted to show that you can do this if you really want to, even thought it is not very useful. Do you think I should just delete this one, and the equivalent syntax examples on the other color function pages?

File: files/en-us/web/css/color_value/color/index.md (hunk 18 above)
Comment (chrisdavidmills) [reply files/en-us/web/css/color_value/color/index.md:50]: OK. What do you think I should say here then? That the value can be written as a number, a percentage, or none. 0-1 represents the bounds of the colorspace, but it is valid to use numbers outside that range? I could then also say that 100% represents 1 and 0% represents 0...is that right? 

In terms of what numbers are useful to use, I'm guessing that depends on which colorspace is used?

File: files/en-us/web/css/color_value/color/index.md (hunk 19 above)
Comment (chrisdavidmills) [reply files/en-us/web/css/color_value/color/index.md:92]: You probably wouldn't, but I felt like it was good for completeness to say that you can do this. Maybe I should update it to say that you can do this, although in reality it is not very useful and you probably wouldn't ever do it. Also maybe say that it isn't really a relative color if you are not using the origin channel values?

File: files/en-us/web/css/color_value/color/index.md (hunk 20 above)
Comment (chrisdavidmills) [reply files/en-us/web/css/color_value/color/index.md:308]: I don't honestly know.

File: files/en-us/web/css/color_value/color/index.md (hunk 17 above)
Comment (mysteryDate) [reply files/en-us/web/css/color_value/color/index.md:25]: I think it might be confusing to the reader. Especially this example with `-1.07e-8`, because the `e` makes it look like something weird might be going on.

File: files/en-us/web/css/color_value/color/index.md (hunk 18 above)
Comment (mysteryDate) [reply files/en-us/web/css/color_value/color/index.md:50]: > In terms of what numbers are useful to use, I'm guessing that depends on which colorspace is used?

Exactly, we've already had a lot of people confused on this point. 
`color(srgb 0 1 0)` is literally just `green1`
`color(display-p3 0 1 0)` is fine for users with p3 displays (which is just about everybody at this point) but will do some potentially surprising gamut mapping for sRGB displays.
`color(xyz 0 1 0)` is _way_ out of gamut for everybody and maybe isn't even a physically valid color.

I like the text you have, I would suggest "but it is valid to use numbers outside that range" -> "values outside of that range will be out of _gamut_ for the given colorspace". I think it's worthwhile to use the technical term since I foresee this causing a lot of confusion.

> I could then also say that 100% represents 1 and 0% represents 0...is that right?

You've discovered one of my least favorite parts of the spec. In this section that is true. It is _not_ true for `lab` or `hsl` or `hwb` where 100% = 100 a lot of the time. It depends on the input range of the parameter. For example, the `l` in `lab` is [0,100], while the `l` in `oklab` is in the range [0,1]. So:
`lab(100% 0.5 0.5)` = `lab(100 0.5 0.5)` and
`oklab(100% 0.5 0.5)` = `oklab(1 0.5 0.5)`

These quirks are due to how colorspaces were defined by the ICC a long time ago, so were stuck with em! I do however feel like the bare numbers for hsl/hwb should be in the range [0,1] not [0,100].

File: files/en-us/web/css/color_value/color/index.md (hunk 19 above)
Comment (mysteryDate) [reply files/en-us/web/css/color_value/color/index.md:92]: Yeah? Maybe one example or note. Idk, I was confused by it.

File: files/en-us/web/css/color_value/color/index.md (hunk 20 above)
Comment (mysteryDate) [reply files/en-us/web/css/color_value/color/index.md:308]: Okay. I'll file one. This has caused enough problems in enough different places by now.

Comment (mysteryDate) [reply files/en-us/web/css/color_value/color/index.md:308]: https://bugs.webkit.org/show_bug.cgi?id=267647

File: files/en-us/web/css/color_value/color/index.md (hunk 17 above)
Comment (chrisdavidmills) [reply files/en-us/web/css/color_value/color/index.md:25]: OK, in my next commit I've gotten rid of this, and other examples that don't really make sense, as discussed.

File: files/en-us/web/css/color_value/color/index.md (hunk 18 above)
Comment (chrisdavidmills) [reply files/en-us/web/css/color_value/color/index.md:50]: Cool, thanks for the detailed explanation here. I have updated the wording to improve accuracy, based on my wording with your suggestion incorporated.

File: files/en-us/web/css/color_value/color/index.md (hunk 19 above)
Comment (chrisdavidmills) [reply files/en-us/web/css/color_value/color/index.md:92]: OK, in each case I have:

- Added a note that reads like so: "Bear in mind that, if you are using relative color syntax but outputting the same color as the origin color or a color not based on the origin color at all, you are not really creating a relative color. You'd be unlikely to ever do this in a real codebase, and would probably just use an absolute color value instead. But, we felt it useful to explain that you _can_ do this with relative color syntax, as a starting point for learning about it."
- Added an extra example that shows how to create a relative color by specifying some of the origin color channel values, but then providing a new value for the others.

File: files/en-us/web/css/color_value/color/index.md (hunk 20 above)
Comment (chrisdavidmills) [reply files/en-us/web/css/color_value/color/index.md:308]: Great, thanks!

**Comment by chrisdavidmills** [comment]
OK, so I've had no more comments on this PR, therefore I'm treating the technical review phase as closed. Thank you for taking the time to review this, everyone!

The next step is to close this PR, and then open a new one against the same branch to contain the editorial review (keeping the two separate ensures cleaner PRs and less confusion due to the sheer weight of comments)
